src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprintAction.java

```diff
@@ -0,0 +1,94 @@
+package org.jenkinsci.plugins.docker.commons.fingerprint;
+
+import hudson.model.Fingerprint;
+import hudson.model.Run;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.CheckForNull;
+import jenkins.model.RunAction2;
+import org.jenkinsci.plugins.docker.commons.Messages;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Adds a link, which provides a list of fingerprints with {@link DockerRunPtrFingerprintFacet}s.
+ * @author Oleg Nenashev
+ */
+public class DockerFingerprintAction implements RunAction2 {
+
+    private final Set<String> fingerprintKeys;
+    transient @CheckForNull Run<?, ?> run;
+
+    public DockerFingerprintAction(Run<?, ?> run) {
+        this.run = run;
+        this.fingerprintKeys = new HashSet<String>();
+    }
+
+    public Run<?, ?> getRun() {
+        return run;
+    }
+    
+    @Override
+    public void onAttached(Run<?, ?> r) {
+        run = r;
+    }
+
+    @Override
+    public void onLoad(Run<?, ?> r) {
+        run = r;
+    }
+
+    @Override
+    public String getIconFileName() {
+        return "fingerprint.png";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return Messages.DockerFingerprintAction_DisplayName();
+    }
+
+    @Override
+    public String getUrlName() {
+        return "docker-fingerprints";
+    }
+
+    public Set<String> getFingerprintKeys() {
+        return fingerprintKeys;
+    }
+    
+    @Restricted(NoExternalUse.class)
+    public @CheckForNull Fingerprint getFingerprint(String imageId) {
+        try {
+            return DockerFingerprints.of(imageId);
+        } catch (IOException ex) {
+            return null; // nothing to do in web UI - return null as well
+        }
+    }
+    
+    /**
+     * Adds an action with a reference to fingerprint if required.
+     * @param fp Fingerprint
+     * @param imageId ID of the docker image
+     * @param run Run to be updated
+     * @throws IOException Cannot save the action
+     */
+    static void addToRun(Fingerprint fp, String imageId, Run run) throws IOException {
+        synchronized (run) {
+            DockerFingerprintAction action = run.getAction(DockerFingerprintAction.class);
+            if (action == null) {
+                action = new DockerFingerprintAction(run);
+                run.addAction(action);
+            }
+
+            if (!action.getFingerprintKeys().contains(imageId)) {
+                action.getFingerprintKeys().add(imageId);
+            } else {
+                return; // no need to save updates
+            }
+
+            run.save();
```

Comment:
If you like, this can be shortened a bit:

```java
if (action.getFingerprintKeys().add(imageId)) {
    run.save();
} // else no need to save updates
```